models/issue.go

```diff
@@ -1055,11 +1055,20 @@ func getMaxIndexOfIssue(e Engine, repoID int64) (int64, error) {
 func newIssue(e *xorm.Session, doer *User, opts NewIssueOptions) (err error) {
 	opts.Issue.Title = strings.TrimSpace(opts.Issue.Title)
 
-	maxIndex, err := getMaxIndexOfIssue(e, opts.Issue.RepoID)
-	if err != nil {
-		return err
+	if opts.Issue.Index == 0 {
+		maxIndex, err := getMaxIndexOfIssue(e, opts.Issue.RepoID)
+		if err != nil {
+			return err
+		}
+		opts.Issue.Index = maxIndex + 1
+	} else if !doer.IsAdmin {
+		// Require admin to specify Issue.Index
+		return ErrUserDoesNotHaveAccessToRepo{UserID: doer.ID, RepoName: opts.Repo.Name}
+	}
+
+	if opts.Issue.Index < 1 {
+		return fmt.Errorf("Issue number out of range or max issue number reached")
```

Comment:
This is not correct way to return error as such messages should be translatable when shown in UI

Reply:
@lafriks Sorry, this is a 500 error outside of the Macaron context... I've tried to find an example in the code for a translatable 500 error but I couldn't. The 500 page itself is not translated. What would be the right way?

![image](https://user-images.githubusercontent.com/18600385/63282593-89368000-c285-11e9-9162-f9c5b73e7137.png)

Reply:
But if it would return fixed error type like `ErrUserDoesNotHaveAccessToRepo` above it could later be checked to return specific error message like it is done here:
https://github.com/go-gitea/gitea/blob/f369788347167a47a8fc162e086b92048ff0a43f/routers/repo/issue.go#L573

Reply:
But it is probably quite an edge case but would be still nice to have

Reply:
@lafriks I don't think this error will ever show up unless someone specifically attempts to break Gitea by uploading issues with an index value close to max(int64) through the API... It truly is a 500 Server Error. Furthermore, the user has no valid course of action upon it; only the site admin (if anyone), which by that moment should have their face all red. Extra work for the translators, not much value. 😁 
If you are OK with it, I'd resolve this conversation.

Reply:
Ok